### ISSUE TITLE
Nodes not properly configured??

### ISSUE THREAD

**Issue by Paul-Reed** [mikejac/node-red-contrib-google-smarthome#123]
Hi, I've had a lot of success with smarthome, thanks for sharing.

Despite **everything appearing to work ok**, I am getting a 'nodes not configured' red triangle on the smarthome nodes.
I am able to control a shelly dimmer's brightness & on/off ok, with a dimmer node, with no debug error messages or anything obvious in the node-RED log.
This animated gif shows everything working as expected, and even home graph is successfully synced OK;

![dimmer](https://user-images.githubusercontent.com/973580/112531634-b6bba780-8d9f-11eb-8915-e7a7fc138093.gif)
...but note the red warning triangle.

This animated gif shows that by simply opening up the node config, making NO changes whatsoever, the triangles disappear & the deploy button becomes active.
But refreshing the browser (F5) and the triangles again re-appear.

![myerror](https://user-images.githubusercontent.com/973580/112730931-3d9d8b00-8f2c-11eb-86bd-89f8f00e812c.gif)

If I make any changes to any other node in the flow when the red triangles are visible, I (as expected) get a popup telling me that the smarthome nodes are not properly configured.

![deploy](https://user-images.githubusercontent.com/973580/112532818-3302ba80-8da1-11eb-9da2-d819d0d6e6ba.JPG)

I don't know if it's relevant, but I found 'notifications = undefined' in the node-RED log??
```
Devices:SetState(): state = {"online":true,"on":false,"brightness":1}
Device:SetState(): key = "online"
Device:SetState(): key = "on"
Device:SetState(): key = "brightness"
HttpActions:reportState(): deviceId = 9220cd97.7bda5
HttpActions:reportState(): states = {"online":true,"on":false,"brightness":1}
HttpActions:reportState(): notifications = undefined
HttpActions:reportState(): postData = {"requestId":"3jWIQ4dDDNBRtFNqddHrRQ","age                                                                             ntUserId":"0","payload":{"devices":{"states":{"9220cd97.7bda5":{"online":true,"o                                                                             n":false,"brightness":1}}}}}
HttpActions:reportState(): success
```

Node-RED version: v1.2.7
Node.js  version: v12.21.0
Linux 5.10.17-v7+ arm LE (Raspberry Pi OS)
google-smarthome v0.0.63

EDIT - Just updated node-RED to v1.2.9 and issue still remains.

**Comment by Caprico85** [comment]
The red triangle is usually triggered by the form field validation which runs in the browser.

Which browser do you use?
Do you have any messages in your browsers JS console?
Does the same happen if you use another browser?
Does the same happen if you create a new management node? You can leave your exisiting management node as is. Just create a second management node and fill in some dummy data.

**Comment by Paul-Reed** [comment]
1) I'm using Google Chrome Version 89.0.4389.90 (Official Build) (64-bit)
2) No there is not any messages in the browser JS console
3) Yes, the same happens in Microsoft Edge Version 89.0.774.63 (Official build) (64-bit)
4) Yes, the same happens if I create a new management node using dummy data, even on a different machine.

I have attached my management node below which contains the dummy data, and the red triangle

```
[{"id":"9aec3ef7.e1de7","type":"google-mgmt","z":"c9c4eeb8.bad8e","client":"7e35b396.77d7ac","name":"Smarthome management","x":640,"y":210,"wires":[[]]},{"id":"7e35b396.77d7ac","type":"googlesmarthome-client","name":"Test","enabledebug":true,"usegooglelogin":true,"loginclientid":"test.apps.googleusercontent.com","emails":"test@gmail.com","username":"","password":"","usehttpnoderoot":true,"port":"2053","httppath":"/","ssloffload":false,"publickey":"/home/pi/.node-red/certs/cert.pem","privatekey":"/home/pi/.node-red/certs/privkey.pem","jwtkey":"/home/pi/.node-red/smarthome.json","accesstokenduration":"60","reportinterval":"60","clientid":"test","clientsecret":"test"}]
```

**Comment by Caprico85** [comment]
Thank you. With your exported mgmt node I am able to reproduce the error. Looking for a fix now.

**Comment by Caprico85** [comment]
Still thinking about how to fix this. But you can work around the problem by unchecking "Use Google Login" in the config, entering some dummy data as username and password, then re-check "Use Google login". Username and password won't be used anywhere. It's just these two fields are incorrectly validated when they are empty.

**Comment by Paul-Reed** [comment]
Yes, it seems to be a validation rule issue.
The existing validations were added by @ckhmer1 - https://github.com/mikejac/node-red-contrib-google-smarthome/commit/e7adf0de378cd33f619d8b8f3069478673727652#diff-df897dc85897b43b5fede0a31f34b3f67f0f6cb45dc75d7890f5232f4369ce8b

**Comment by ckhmer1** [comment]
Hi,
I'm sorry but I'm not able to reproduce the problem.
I've imported the flow, saved it, and no red triangle.
I've checked the validation rules and seem correct to me.

**Comment by Paul-Reed** [comment]
That's strange because I've tried it on two separate systems, and also @Caprico85 appears to have reproduced the error.
I'm not very advanced at programming/debugging, but can you suggest any tests that I can do to try and help pin this issue down?

**Comment by Caprico85** [comment]
It should be reproducable when you disable "Use Google Login", then delete username and password, then re-enable "Use Google Login", deploy and finally reload the browser window. If that doesn't help, try re-deploying or opening and closing the config node a few times without changing anything.

I'm still not sure what exactly causes this problem. I noticed that during the browser reload the validation already runs before the DOM is completely loaded. So in the validation rule, `$("#node-config-input-usegooglelogin")` doesn't find the element in the DOM. But this should already be handled in the else branch. Not sure what else could cause the problem.

**Comment by Caprico85** [comment]
Fixed in v0.0.68

Problem was that the validation functions tried to access the other form fields (e.g. password validation check if usegooglelogin is set). But validation is also called during initialization of Node-Red, when the other form fields don't exist yet.
